### DISCUSSION TITLE
Can wifi work? -- a hint on wifi file uploads

### DISCUSSION THREAD

**Discussion by nathanielstenzel** [aegean-odyssey/mpmd_marlin_1.1.x#85] [General]
"Transfers happen over telnet, which blocks the sending of any other GCode commands and limits how fast the files can be transferred" https://github.com/nokemono42/MP-Mini-Delta-Web

**Comment by aegean-odyssey** [comment]
___The wifi feature is disabled in `mpmd_marlin_1.1.x` firmware___. The wifi feature never seemed to work too well for me with the stock firmware, and I never got it to work reliably enough in the `mpmd_marlin_1.1.x` firmware to warrant its inclusion. Since the "heavy lifting" for the wifi is built into the UI board (not the controller board where the firmware is installed), many issues with the wifi are issues with its implementation on the UI board -- outside of the scope of this project.

There was a bit of effort to include the wifi feature, but these main issues with the wifi made including the feature more or less intractable
* the telnet connection on the UI board did not handle commands reliably with more than a small amount of output;
* an unreliable or busy wifi connection directly impacted the printer's overall performance (processing gcode, motion control, etc.);
* the "file upload" protocol used between the controller and UI boards is undocumented, a huge obstacle to implementing it.

Effort/reward/cost -wise, I think adding a host on such as OctoPrint via the usb port offers more opportunities to improve the overall "MP Mini Delta" printer system. Truly working WIFI, cameras, lights, better user interfaces, ... are all possible by adopting this route to expanding and improving the printer.

**Comment by nathanielstenzel** [comment]
Ah. Well, I just figured I would give you that link since I could have sworn I saw somewhere on here that you said you could not figure it out. I could not find it just now when I looked but I have not woken up all the way either. I figure that you could at least add it to your notes for in case anyone else cares to bother. Meanwhile, I agree that WIFI uploads were totally horrible. I do think that the system could potentially be tricked into working better with checksums, small file chunks and such, but that as you said would be a bit of effort.

**Comment by nathanielstenzel** [comment]
I guess if nothing else, by making this issue post, I have added it to your notes for future reference for whoever may care.

**Comment by aegean-odyssey** [comment]
Yes, the link is interesting -- it's the other end of the wifi problem: the project modifies the "web pages" that reside in the UI board for a better web experience. It's a definite improvement, but it can't overcome the fundamental shortcomings with the communication between the mini delta's UI and controller boards.
